loaders/src/glTF/2.0/Extensions/MSFT_audio_emitter.ts

```diff
@@ -0,0 +1,459 @@
+                                    if (emitter.innerAngle != undefined || emitter.outerAngle != undefined) {
+                                        sound.setLocalDirectionToMesh(new Vector3(0, 0, 1));
+                                        sound.setDirectionalCone(2 * Tools.ToDegrees(emitter.innerAngle == undefined ? Math.PI : emitter.innerAngle),
+                                                                 2 * Tools.ToDegrees(emitter.outerAngle == undefined ? Math.PI : emitter.outerAngle), 0);
+                                    }
+                                }
+                            }
+                        }));
+                    }
+
+                    return Promise.all(promises).then(() => {});
+                });
+            });
+        }
+
+        protected _loadAnimationAsync(context: string, animation: _ILoaderAnimation): Nullable<Promise<void>> { 
+            return this._loadExtensionAsync<_ILoaderAnimationEvents>(context, animation, (extensionContext, extension) => {
+                return this._loader._loadAnimationAsync(extensionContext, animation).then(() => {
+                    const promises = new Array<Promise<void>>();
+                    let babylonAnimationGroup = animation._babylonAnimationGroup;
+
+                    _ArrayItem.Assign(extension.events);
+                    for (const event of extension.events) {
+                        promises.push(this._loadAnimationEventAsync(`${context}/extension/${NAME}/events/${event._index}`, context, animation, event, babylonAnimationGroup!));
+                    }
+
+                    return Promise.all(promises).then(() => {
+                        // Make sure all audio stops when the animation is terminated.
+                        const getCallback = (pause: Boolean) => {
+                            const emitterList = this._emitters;
+                            return () => { 
+                                for (const emitter of emitterList) {
+                                    if (pause) {
+                                        emitter._babylonData!.sound!.pause();
+                                    } else {
+                                        emitter._babylonData!.sound!.stop();
+                                    }
+                                }
+                            };
+                        }
+                        babylonAnimationGroup!.onAnimationGroupEndObservable.add(getCallback(false));
+                        babylonAnimationGroup!.onAnimationGroupPauseObservable.add(getCallback(true));
+    
+                        babylonAnimationGroup!.normalize();
+                    });
+                });
+            });
+        }
+
+        private _getEventAction(sound: WeightedSound, action: _AnimationEventAction, time: number, startOffset?: number): (currentFrame: number) => void {
+            return (currentFrame: number) => {
```

Comment:
it would be more efficient if you check the action outside of the lambda instead